src/main/java/io/cryostat/recordings/RecordingMetadataManager.java

```diff
@@ -377,19 +379,23 @@ else if (fs.isRegularFile(subdirectory)) {
                             "Event bus [{}]: {}",
                             DiscoveryStorage.DISCOVERY_STARTUP_ADDRESS,
                             message.body());
-                    vertx.executeBlocking(
-                            promise -> {
-                                try {
-                                    archiveHelper.migrate();
-                                    logger.info("Successfully migrated archives");
-                                    pruneStaleMetadata(staleMetadata);
-                                    logger.info("Successfully pruned all stale metadata");
-                                    promise.complete();
-                                } catch (Exception e) {
-                                    logger.warn("Couldn't read archived recordings directory...");
-                                    promise.fail(e);
-                                }
-                            });
+                    new Thread(
```

Comment:
Good question. This was just to ensure that the new CountDownLatch can't cause a deadlock or thread starvation. If there is only one worker thread in the pool and we call `executeBlocking` at the end of `start()`, as well as in `accept(TargetDiscoveryEvent)`, then it could be possible [0] for the worker in the pool to start handling the `accept` first, which will cause the worker thread to block and await the latch counting down. The unit of work that would actually count down the latch would be stuck in the work queue behind the blocked task, and the worker thread would never go perform that work to unblock itself.

[0] We currently have a defined startup ordering between the DiscoveryStorage and the RecordingMetadataManager that *should* ensure that `start()` is called and the unit of work that releases the latch is entered into the work queue first for the worker thread to begin on, before any `accept()` could be called, but that relationship could change and is not controlled by the metadata manager class. So this may not be strictly required now but t's a defensive move to prevent deadlocking our own worker thread pool.